iiif-data-service.js

```diff
@@ -18,42 +18,42 @@ export class IIIFDataService {
 
     /**
      * Fetch and parse IIIF canvas data with annotations
-     * @param {string} canvasUrl - URL to the IIIF canvas or manifest containing the canvas
+     * @param {string} pageId - ID of the IIIF page to load
      * @returns {Promise<Object|null>} Canvas data with image URL, annotations, and dimensions
      */
-    async fetchCanvasData(canvasUrl) {
+    async fetchPageData(pageId, manifestUrl = null) {
         try {
-            const response = await fetch(canvasUrl)
+            const response = await fetch(manifestUrl || pageId)
             if (!response.ok) {
                 throw new Error(`Failed to fetch data from URL: ${response.status}`)
             }
 
             const data = await response.json()
-            
+
             // Check if this is manifest data that contains canvases
             if (this.isManifestData(data)) {
-                return await this.extractCanvasFromManifest(data, canvasUrl)
+                const pageData = await fetch(pageId)
```

Comment:
Do we never want to use the AnnotationPage embedded in the Manifest's Canvas?  It could save us a fetch().